test/e2e/mcd_test.go

```diff
@@ -92,57 +86,70 @@ func TestMCDeployed(t *testing.T) {
 			},
 			Storage: ignv2_2types.Storage{
 				Files: []ignv2_2types.File{
-					createMCFile("/etc/mytestconf", "data:,test", 420),
+					createIgnFile(filename, "data:,"+data, 420),
 				},
 			},
 		},
 	}
+	return mcadd
+}
 
-	// create the dummy MC now
-	_, err = mcClient.MachineconfigurationV1().MachineConfigs().Create(mcadd)
+func TestMCDeployed(t *testing.T) {
+	cb, err := common.NewClientBuilder("")
 	if err != nil {
-		t.Errorf("failed to create machine config %v", err)
+		t.Errorf("%#v", err)
 	}
+	mcClient := cb.MachineConfigClientOrDie("mc-file-add")
+	k := cb.KubeClientOrDie("mc-file-add")
 
-	// grab the latest worker- MC
-	var newMCName string
-	if err := wait.PollImmediate(2*time.Second, 5*time.Minute, func() (bool, error) {
-		mcp, err := mcClient.MachineconfigurationV1().MachineConfigPools().Get("worker", metav1.GetOptions{})
-		if err != nil {
-			return false, err
-		}
-		for _, mc := range mcp.Status.Configuration.Source {
-			if mc.Name == mcName {
-				newMCName = mcp.Status.Configuration.Name
-				return true, nil
-			}
-		}
-		return false, nil
-	}); err != nil {
-		t.Errorf("machine config hasn't been picked by the pool: %v", err)
-	}
+	for i := 0; i < 10; i++ {
+		mcadd := createMCToAddFile("00-0add-a-file", fmt.Sprintf("/etc/mytestconf%d", i), "test")
```

Comment:
doesn't the `00-0add-a-file` name need to change here as well? since it's passed to `createMCToAddFile` and runs into the same problem?

Reply:
the issue before was that there wasn't anything other than a random UID, with this now it's that name (which makes it to the top of the list) + a random hash just to differentiate from the others

Reply:
like just `add-a-file` so the name genereated in createMCToAddFile will be `add-a-file-XXXXXXXXX`?

Reply:
I can do that sure, not sure if it changes something though :thinking:

Reply:
i thought the worry was that the uuid would sort before the build in MCs. but if we use 00-0-add... won't this sort before the built-in MCs now every time?

Reply:
oooo gotcha finally, you're completely right, changing

Reply:
fixed